### ISSUE TITLE
Bit order not specified in <8bpp blitters

### ISSUE THREAD

**Issue by ccawley2011** [libsdl-org/SDL#6242]
The blitters in `src/video/SDL_blit_0.c` don't seem to attempt to distinguish between `SDL_PIXELFORMAT_INDEX1LSB` and `SDL_PIXELFORMAT_INDEX1MSB`, or  `SDL_PIXELFORMAT_INDEX4LSB` and `SDL_PIXELFORMAT_INDEX4MSB`, which potentially results in incorrect conversion to higher bit depths.

**Comment by slouken** [comment]
Yup, good point. Feel free to submit a tested PR to fix this. Ideally you'd update the automated tests to cover this case.

**Comment by ccawley2011** [comment]
Am I correct in thinking that the current blitters are correct for MSB formats?

**Comment by slouken** [comment]
I don't know, can you write a quick test case to find out?

**Comment by slouken** [comment]
Yes, I confirmed the current blitters assume MSB format.
